### ISSUE TITLE
Load gallery from ajax loaded content

### ISSUE THREAD

**Issue by Teufeuleu** [nanostudio-org/nanogallery2#178]
Hi,
The website i'm working on is based on an index.html containing a menu that basically loads external .html contents into a .container div through AJAX requests.

The whole gallery(ies) I want to load are contained into external .html. Galleries are simple : grid layout with available lightbox, only self-hosted images, no checkboxes, no album...
Loading works well if I use the [javascript method](https://nanogallery2.nanostudio.org/datasource.html#ngy2_define_javascript), but it does not if I use the html markup.

I bet it's because nanogallery2 loading functions are not called when the external .html is loaded, but I cannot find any way to do it.

I saw refresh and reload methods in the API but obviously none of them work, as no nanogallery2 instance exists. I also naively tried to call `$("#myGallery).nanogallery2();` just after loading my external .html but it does not work.

So in a nutshell what i'm trying to do is dynamically add or remove galleries from ajax loaded .html while keep using simple html markdup for convenience.

Just to be clear : I DO NOT want to dynamically add/remove items from an existing gallery, but load/delete galleries embedded into dynamically loaded html content.

And thanks for this amazing work !

**Comment by Kris-B** [comment]
Hi, you are right, nanogallery2 will check setting in html markup on first page load.
You can force this behaviour after loading an external page:

```
      var t=document.querySelectorAll('[data-nanogallery2]');
      for( var i=0; i < t.length; i++ ) {
        jQuery( t[i] ).nanogallery2( jQuery(t[i]).data('nanogallery2') );
      }
```

**Comment by Kris-B** [comment]
@Teufeuleu any update ?
